bundles/specmate-emfrest/src/com/specmate/emfrest/authentication/AuthRestServiceBase.java

```diff
@@ -0,0 +1,69 @@
+package com.specmate.emfrest.authentication;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
+
+import com.specmate.common.exception.SpecmateException;
+import com.specmate.emfrest.api.RestServiceBase;
+import com.specmate.emfrest.internal.auth.AuthorizationHeader;
+import com.specmate.rest.RestResult;
+import com.specmate.usermodel.User;
+import com.specmate.usermodel.UserSession;
+
+public abstract class AuthRestServiceBase extends RestServiceBase implements IResponseAlteringService {

```

Comment:
Please add a comment on what the class does. Especially: When to use this as a base class and when use only RestServiceBase